specgen/specgen/main.go

```diff
@@ -0,0 +1,117 @@
+// Command specgen provides a utility for generating utls.ClientHelloSpecs. This command starts an
+// HTTPS server which responds to every request with a ClientHelloSpec corresponding to the TLS
+// ClientHello the server received. The spec is logged for every request as well. Make a request of
+// the server via web browser to see a ClientHelloSpec for the browser.
+//
+// It is recommended that mkcert be used to generate the keypair for this server. This ensures
+// browsers do not flag and block the page. See https://github.com/FiloSottile/mkcert.
+package main
+
+import (
+	"bytes"
+	"crypto/tls"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/getlantern/utls/specgen"
+	"github.com/getlantern/utls/specgen/specgen/hellocap"
+)
+
+const certFileDefaultText = "pre-defined, self-signed cert"
+
+var (
+	addr          = flag.String("addr", "localhost:0", "address to listen on")
+	certFile      = flag.String("cert", certFileDefaultText, "PEM-encoded TLS certificate file")
+	keyFile       = flag.String("key", "", "PEM-encoded TLS key file")
+	packagePrefix = flag.String("prefix", "", "package prefix; see specgen.WriteHelloSpec")
```

Comment:
Looking in a library package to find the documentation for a binary command would frustrate me (as a developer using this).

Reply:
Yeah, that's fair.  I'm not sure how to condense the explanation into something which fits in a flag usage string though.  Any ideas?  There are alternative CLI and/or flag libraries, but that seems like a big hammer for this issue.

Reply:
`--tls-package: if set, replaces the 'tls' in e.g. 'tls.ClientHelloInfo' in the generated code. By default, no prefix is used.`